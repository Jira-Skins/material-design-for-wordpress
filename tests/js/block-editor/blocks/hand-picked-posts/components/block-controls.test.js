/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
/**
 * Internal dependencies
 */
import HandPickedPostsBlockControls from '../../../../../../assets/src/block-editor/blocks/hand-picked-posts/components/block-controls';

/**
 * Render the component.
 *
 * @param {Object} props - Component props
 * @return {Function} A functional component.
 */
const setup = props => {
	return shallow( <HandPickedPostsBlockControls { ...props } /> );
};

describe( 'HandPickedPostsBlockControls', () => {
	it( 'matches snapshot when the edit mode is true', () => {
		const wrapper = setup( {
			attributes: { editMode: true },
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when the edit mode is false', () => {
		const wrapper = setup( {
			attributes: { editMode: false },
		} );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
