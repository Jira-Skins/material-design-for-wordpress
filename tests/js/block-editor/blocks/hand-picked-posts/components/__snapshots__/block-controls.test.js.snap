// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandPickedPostsBlockControls matches snapshot when the edit mode is false 1`] = `
<IfBlockEditSelected(BlockControlsFill)>
  <Toolbar
    controls={
      Array [
        Object {
          "icon": "edit",
          "isActive": false,
          "onClick": [Function],
          "title": "Edit",
        },
      ]
    }
  />
</IfBlockEditSelected(BlockControlsFill)>
`;

exports[`HandPickedPostsBlockControls matches snapshot when the edit mode is true 1`] = `
<IfBlockEditSelected(BlockControlsFill)>
  <Toolbar
    controls={
      Array [
        Object {
          "icon": "edit",
          "isActive": true,
          "onClick": [Function],
          "title": "Edit",
        },
      ]
    }
  />
</IfBlockEditSelected(BlockControlsFill)>
`;
